src/shell-util.c

```diff
@@ -492,3 +499,117 @@ shell_util_composite_capture_images (ClutterCapture  *captures,
 
   return image;
 }
+
+static void
+find_latin_keysym (struct xkb_keymap *keymap,
+                   xkb_keycode_t      key,
+                   void              *data)
+{
+  FindLatinKeysymsState *state = data;
+  int n_keysyms, i;
+  const xkb_keysym_t *keysyms;
+
+  n_keysyms = xkb_keymap_key_get_syms_by_level (keymap,
+                                                key,
+                                                0,
+                                                0,
+                                                &keysyms);
+  for (i = 0; i < n_keysyms; i++)
+    {
+      xkb_keysym_t keysym = keysyms[i];
+      if (keysym >= XKB_KEY_a && keysym <= XKB_KEY_z)
+        {
+          unsigned int keysym_index = keysym - XKB_KEY_a;
+          if (!state->required_keysyms_found[keysym_index])
+            {
+              state->required_keysyms_found[keysym_index] = TRUE;
+              state->n_required_keysyms--;
+            }
+        }
+    }
+}
+
+static struct xkb_keymap *
+create_keymap(struct xkb_context *context,
+              const char         *layouts,
+              const char         *variants,
+              const char         *options)
+{
+    struct xkb_keymap *keymap;
+    struct xkb_rule_names rmlvo = {
+        .rules = "evdev",
+        .model = "pc104",
```

Comment:
Is it a good idea for these two strings to be hardcoded? You could use `#defines` if so.

Reply:
In Mutter they are defined as evdev and pc105(pc105+inet).

https://github.com/GNOME/mutter/blob/master/src/backends/meta-backend-private.h#L48

https://github.com/GNOME/mutter/blob/master/clutter/clutter/evdev/clutter-device-manager-evdev.c#L1970